### PR TITLE
Set esm2017 bundles as default for exp packages

### PR DESCRIPTION
**Pull request by hsubox76**
esm2017 bundles will be default in the `browser` and `module` fields for exp and compat packages.
A new `esm5` field will be added for users to get the esm5 bundle if needed.

### CODE REVIEW

**Review by Feiyang1 — APPROVED** [review]

File: common/api-review/firestore.api.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
+## API Report File for "@firebase/firestore"
```
Comment (Feiyang1) [inline common/api-review/firestore.api.md:1]: This seems unrelated to the PR.

File: packages-exp/analytics-compat/rollup.shared.js [hunk 2]
```diff
@@ -45,7 +45,7 @@ export const es2017BuildsNoPlugin = [
      */
     input: 'src/index.ts',
     output: {
-      file: pkg.esm2017,
+      file: pkg.browser,
```
Comment (Feiyang1) [inline packages-exp/analytics-compat/rollup.shared.js:48]: I can't put comments for the es5 build because it's not part of the PR - `pkg.module` should point to the esm2017 build, and `pkg.esm5` should be used to create the es build.

**Review by Feiyang1 — CHANGES_REQUESTED** [review]

File: packages-exp/analytics-compat/rollup.shared.js (hunk 2 above)
Comment (hsubox76) [reply packages-exp/analytics-compat/rollup.shared.js:48]: Fixed.

File: common/api-review/firestore.api.md (hunk 1 above)
Comment (hsubox76) [reply common/api-review/firestore.api.md]: Removed.

File: packages/firestore/compat/package.json [hunk 3]
```diff
@@ -8,8 +8,7 @@
     "react-native": "../dist/compat/rn/index.js",
     "browser": "../dist/compat/esm2017/index.js",
     "module": "../dist/compat/esm2017/index.js",
-    "esm2017": "../dist/compat/esm2017/index.js",
     "license": "Apache-2.0",
-    "typings": "../dist/compat/esm5/packages/firestore/compat/index.d.ts"
+    "typings": "../dist/compat/esm2017/packages/firestore/compat/index.d.ts"
```
Comment (Feiyang1) [inline packages/firestore/compat/package.json:12]: Should we have a esm5 field here?

File: packages/storage/compat/package.json [hunk 4]
```diff
@@ -8,6 +8,6 @@
     "module": "../dist/compat/esm5/index.js",
     "esm2017": "../dist/compat/esm2017/index.js",
```
Comment (Feiyang1) [inline packages/storage/compat/package.json:9]: We need to update the fields and rollup.config for storage exp and compat.

**Review by Feiyang1 — APPROVED** [review]
